src/PostgREST/ApiRequest.hs

```diff
@@ -181,10 +183,18 @@ userApiRequest schema req reqBody =
   tokenStr = case T.split (== ' ') (cs auth) of
     ("Bearer" : t : _) -> t
     _                  -> ""
-  endingIn:: T.Text -> T.Text -> Bool
-  endingIn word key = word == lastWord
+  endingIn:: [T.Text] -> T.Text -> Bool
+  endingIn xx key = lastWord `elem` xx
     where lastWord = last $ T.split (=='.') key
 
+  headerRange = if singular then Just (singletonRange 0) else rangeRequested hdrs
+  urlRange = toRange (join $ lookup "limit" qParams) (join $ lookup "offset" qParams)
+
+  setTopRange :: Maybe NonnegRange -> M.HashMap String NonnegRange -> M.HashMap String NonnegRange
```

Comment:
This name confused me because I thought it was referring to limiting the upper value of a range (the range's "top value") but I guess it refers to ranges for top levels of the query tree you're building?